### PR TITLE
Unwrap reduction 2bis and versions bump

### PR DESCRIPTION
**Pull request by redmie**
Depends on MR #17.
This supplant MR #14 which must be closed without being merged. 

This MR continues the work of removing unwrap form the code.

**WARNING: Merge #17  before this one**

### CODE REVIEW

**Comment by redmie** [comment]
Can you review that MR please @dobefore

**Comment by dobefore** [comment]
While running server(built for Windows) and trying to sync,error occurs.
```
[2022-03-14T06:38:51Z INFO  actix_server::builder] Starting 4 workers
[2022-03-14T06:38:51Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
Authentication succeeded for  user pp.
[2022-03-14T06:39:48Z INFO  actix_web::middleware::logger] 192.168.0.88 "POST /sync/hostKey HTTP/1.1" 200 42 "-" "AnkiDroid-2.15.6" 0.037309
```
```
thread 'actix-rt|system:0|arbiter:2' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', C:\Users\Admin\.cargo\registry\src\github.com-1ecc6299db9ec823\tokio-1.16.1\src\runtime\enter.rs:39:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Is this a probem with actix update?

**Comment by redmie** [comment]
I didn't catch that! Thanks @dobefore.

**Comment by dobefore** [comment]
yes,of course.
```
stack backtrace:
   0:     0x7ff70591fecf - std::backtrace_rs::backtrace::dbghelp::trace
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
   1:     0x7ff70591fecf - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff70591fecf - std::sys_common::backtrace::_print_fmt
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\sys_common\backtrace.rs:67
   3:     0x7ff70591fecf - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\sys_common\backtrace.rs:46
   4:     0x7ff70593913a - core::fmt::write
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\core\src\fmt\mod.rs:1163
   5:     0x7ff705917ba8 - std::io::Write::write_fmt<std::sys::windows::stdio::Stderr>
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\io\mod.rs:1696
   6:     0x7ff705922b06 - std::sys_common::backtrace::_print
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\sys_common\backtrace.rs:49
   7:     0x7ff705922b06 - std::sys_common::backtrace::print
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\sys_common\backtrace.rs:36
   8:     0x7ff705922b06 - std::panicking::default_hook::closure$1
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\panicking.rs:210
   9:     0x7ff705922507 - std::panicking::default_hook
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\panicking.rs:227
  10:     0x7ff705923165 - std::panicking::rust_panic_with_hook
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\panicking.rs:624
  11:     0x7ff7058d55d5 - std::thread::local::fast::Key<T>::try_initialize::hbeb72e9db473ae42
  12:     0x7ff7058d50a7 - std::sys_common::backtrace::__rust_end_short_backtrace::h590887c7ac505bd5
  13:     0x7ff705af8c3d - std::panicking::begin_panic::hb204d80bf0aae628
  14:     0x7ff7058c6d33 - tokio::runtime::enter::enter::h3f277f6fe78c2401
  15:     0x7ff7056e8d97 - tokio::runtime::thread_pool::ThreadPool::block_on::h3d4326c204dffbe4
  16:     0x7ff70572b9c5 - tokio::runtime::Runtime::block_on::h129eb8ad2df74acd
  17:     0x7ff70564dc21 - anki::backend::sync::server::<impl anki::backend::Backend>::meta::h12d086c5db382608
  18:     0x7ff70565203e - anki::backend::sync::server::<impl anki::backend::Backend>::sync_server_method_inner::h56d2f3d1601ec906
  19:     0x7ff705652e0d - anki::backend::sync::<impl anki::backend_proto::sync::sync_service::Service for anki::backend::Backend>::sync_server_method::h2589ae9c38e517c9
  20:     0x7ff705453bd1 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hfbff2dfe08fafacd
  21:     0x7ff70544a8f7 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h87359057f95808d5
  22:     0x7ff705449211 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h18726c69d6a6e48d
  23:     0x7ff7053da74d - <actix_web::middleware::logger::LoggerResponse<S,B> as core::future::future::Future>::poll::hbffd4a016fe003b0
  24:     0x7ff7053f99a8 - <actix_service::map_err::MapErrFuture<A,Req,F,E> as core::future::future::Future>::poll::h90a5e50199145655
  25:     0x7ff7053ba49a - actix_http::h1::dispatcher::InnerDispatcher<T,S,B,X,U>::poll_response::h0b53356c1c114a92
  26:     0x7ff7053fa01e - <actix_web::service::ServiceFactoryWrapper<T> as actix_web::service::AppServiceFactory>::register::hc843e6b670efc69e
  27:     0x7ff7053fbba3 - <actix_http::service::HttpServiceHandlerResponse<T,S,B,X,U> as core::future::future::Future>::poll::h8fa7d0be31c33fd0
  28:     0x7ff705466c91 - <actix_service::and_then::AndThenServiceResponse<A,B,Req> as core::future::future::Future>::poll::h8092669152aead21
  29:     0x7ff705449d77 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h670e7fcf7ad11338
  30:     0x7ff705496bec - std::panicking::try::h5bd325e6f6a3ffad
  31:     0x7ff705431f4d - tokio::runtime::task::harness::Harness<T,S>::poll::h1234ca9c58f53c28
  32:     0x7ff7058d28fe - std::thread::local::LocalKey<T>::with::hfe7e8ce59d6aab61
  33:     0x7ff7058bacec - tokio::task::local::LocalSet::tick::h2d55f9615d2ba36c
  34:     0x7ff7055242c9 - tokio::macros::scoped_tls::ScopedKey<T>::set::haa4a807aee995112
  35:     0x7ff70551b1d5 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::he429e6c331379d65
  36:     0x7ff70551077a - tokio::runtime::basic_scheduler::Context::enter::hc69b3f903a6fef76
  37:     0x7ff70552397b - tokio::macros::scoped_tls::ScopedKey<T>::set::h42b7ae6c65f6a7bc
  38:     0x7ff70550f3f6 - tokio::runtime::basic_scheduler::BasicScheduler::block_on::h28b722f915a39e98
  39:     0x7ff705524de3 - tokio::runtime::Runtime::block_on::h808133c01969c5db
  40:     0x7ff70550bdd8 - std::sys_common::backtrace::__rust_begin_short_backtrace::hfd5b5fae6963317e
  41:     0x7ff7054f9c8b - std::thread::Builder::spawn::hf1e055231605186f
  42:     0x7ff70592601c - alloc::boxed::impl$44::call_once
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\alloc\src\boxed.rs:1691
  43:     0x7ff70592601c - alloc::boxed::impl$44::call_once
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\library\alloc\src\boxed.rs:1691
  44:     0x7ff70592601c - std::sys::windows::thread::impl$0::new::thread_start
                               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c\/library\std\src\sys\windows\thread.rs:58
  45:     0x7ffee5a454e0 - BaseThreadInitThunk
  46:     0x7ffee6b2485b - RtlUserThreadStart

```

**Comment by dobefore** [comment]
Abki seems use tokio::runtime to impl sync functions,such as fn meta() where bock_on() is used.It;s weird!

**Comment by redmie** [comment]
@dobefore I just fixed the runtime in runtime error (it syncs on my test), please test again!the merge conflict

**Comment by dobefore** [comment]
thanks for hard work.As for anki update script,how about using python script for cross-plaform usage?

File: src/sync.rs [hunk 1]
```diff
@@ -10,8 +10,13 @@ use anki::{
     backend::Backend,
     backend_proto::{sync_server_method_request::Method, sync_service::Service},
     media::sync::{
-        slog::{self, o},
-        zip, BufWriter, FinalizeRequest, FinalizeResponse, RecordBatchRequest, SyncBeginResponse,
+        //slog::{self, o},
```
Comment (dobefore) [inline src/sync.rs:13]: I will rmeove this comment in MR #27

**Review by dobefore — APPROVED** [review]

**Pull request merged**